File: modules/common-model/src/main/java/com/shterneregen/securelan/common/model/AppEvent.java
package com.shterneregen.securelan.common.model;

import java.time.Instant;
import java.util.Objects;

public record AppEvent(
        AppEventType type,
        String message,
        Instant createdAt
) {
    public AppEvent {
        Objects.requireNonNull(type, "type must not be null");
        Objects.requireNonNull(message, "message must not be null");
        Objects.requireNonNull(createdAt, "createdAt must not be null");
        if (message.isBlank()) {
            throw new IllegalArgumentException("message must not be blank");
        }
    }
}

File: modules/common-model/src/main/kotlin/com/shterneregen/securelan/common/model/AppEvent.kt
package com.shterneregen.securelan.common.model

import java.time.Instant
import java.util.Objects

@JvmRecord
data class AppEvent(
    val type: AppEventType,
    val message: String,
    val createdAt: Instant,
) {
    init {
        Objects.requireNonNull(type, "type must not be null")
        Objects.requireNonNull(message, "message must not be null")
        Objects.requireNonNull(createdAt, "createdAt must not be null")
        require(message.isNotBlank()) { "message must not be blank" }
    }
}
